lib/internal/Magento/Framework/Css/PreProcessor/Instruction/MagentoImport.php

```diff
@@ -108,9 +119,16 @@ protected function replace(array $matchedContent, LocalInterface $asset)
             $importFiles = $this->fileSource->getFiles($this->getTheme($relatedAsset), $resolvedPath);
             /** @var $importFile \Magento\Framework\View\File */
             foreach ($importFiles as $importFile) {
+                $moduleName = $importFile->getModule();
+
+                if ($moduleName &&
+                    !$this->moduleManager->isEnabled($moduleName)) {
+                    continue;
+                }
```

Comment:
i'm not sure because in the case we disabled module. All js, template related related disabled modules will not render output to html. Only downside of this is can possible happen specific use-cases when disable styling from module A. But in style module A contain some code reference to another component from module B. Disable A may cause some broken visual in module B. Since styles in less from magento quite open and not any restrict you can put styles anywhere you want. In that case we have to perform refactor to migrate styles from A to B

Reply:
Currently i'm not clearly about new implement as you mentioned but i think we should have new flag or config for such case

Reply:
@mrtuvn,
> Since styles in less from magento quite open and not any restrict you can put styles anywhere you want. In that case we have to perform refactor to migrate styles from A to B

Right, but that's also will affect all the 3rt party modules, that's why it's so important to make it configurable. For the Magento core itself, it's much less relevant.

> Currently i'm not clearly about new implement as you mentioned

You should just add a new flag, and check it in the deployment config, similar to https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Config/Model/Config/Structure/ElementVisibility/ConcealInProductionWithoutScdOnDemand.php#L58-L61